src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -802,6 +828,64 @@ private static class ImageHelper implements ImageProvider {
                                 .collect(Collectors.toSet());
         }
 
+        private static Platform targetPlatform(Configuration cf, Map<String, Path> modsPaths) throws IOException {
+            Path javaBasePath = modsPaths.get("java.base");
+            assert javaBasePath != null : "java.base module path is missing";
+            if (isJavaBaseFromCurrentPlatform(javaBasePath)) {
+                // this implies that the java.base module used for the target image
+                // will correspond to the current platform. So this isn't an attempt to
+                // build a cross-platform image. We use the current platform's endianness
+                // in this case
+                return Platform.runtime();
+            } else {
+                // this is an attempt to build a cross-platform image. We now attempt to
+                // find the target platform's arch and thus its endianness from the java.base
+                // module's ModuleTarget attribute
+                String targetPlatformVal = readJavaBaseTargetPlatform(cf);
+                try {
+                    return Platform.parsePlatform(targetPlatformVal);
```

Comment:
You may want to modify `Platform.parsePlatform(s)` to use `Architecture.lookupByName(s)` instead of doing the alias mapping itself.  Platform.java:53-56.

Reply:
That's a good point. I've updated the PR with your suggested implementation.